.github/workflows/build.yml

```diff
@@ -14,37 +36,53 @@ env:
   CCACHE_KEY_SUFFIX: r1
 
 jobs:
-  find-all-packages:
+  generate-matrices:
     runs-on: ubuntu-22.04
     outputs:
-      packages: ${{ steps.list-packages.outputs.packages }}
+      packages-with-dependents: ${{ steps.list-packages-with-dependents.outputs.packages }}
     steps:
       - name: checkout
         uses: actions/checkout@v4
+        with:
+          # Fetch whole git history.
+          # Required to parse modified packages
+          fetch-depth: 0
 
       - name: setup python 3.12
         uses: actions/setup-python@v5
         with:
           python-version: 3.12
       - name: install python dependencies
-        run: pip install --upgrade pip pyyaml
+        run: pip install --upgrade pip conan
+
+      - name: conan config
+        run: conan config install .github/config/ubuntu-22.04/conan
```

Comment:
maybe we can replace `ubuntu-22.04` by a github variable here

Reply:
I've tried to set `ubuntu-22.04` as env var and use it in `runs-on` and in `conan install` path, but I've found out that we can't have env vars as inputs to `runs-on`  ( https://github.com/opendocument-app/conan-odr-index/actions/runs/9847725749 )

`conan install` path can use env var, but we still define `ubuntu-22.04` in two places. Unless I do a one entry matrix, because `runs-on` can read from matrices, but I doubt that would look better than it looks now.